### PR TITLE
NEW: TripSector and TripSegment component

### PR DESCRIPTION
**Pull request by mainframev**
This Pull Request meets the following criteria:

* [ ] Tests have been added/adjusted for my new feature
* [ ] New Components are registered in index.js of my project
<br/><br/><br/><url>LiveURL: https://orbit-components-new-TripSector.surge.sh</url>

### CODE REVIEW

**Comment by wEprous** [comment]
Done. @tomashapl can you review it?

File: src/TripSegment/index.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    case "bus":
+      return <Bus size="small" color="secondary" />;
+    default:
+      return <Airplane size="small" color="secondary" />;
+  }
+};
+
+class TripSegment extends React.PureComponent<Props, State> {
+  state = {
+    toggle: this.props.shown || false,
```
Comment (tomashapl) [inline src/TripSegment/index.js]: Naming is hard i know, but this is little bit confusing.... 

please use `expanded` and `initialExpanded`

File: src/TripSegment/index.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    visible: !this.props.shown,
+  };
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setHeight();
+    }, 500);
+  }
+
+  setHeight() {
```
Comment (tomashapl) [inline src/TripSegment/index.js]: We can do it in more simply way:

image


image

File: src/TripSegment/index.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+      toggle: !prevState.toggle,
+      visible: true,
+    }));
+
+    if (onClick) {
+      onClick();
+    }
+  };
+
+  contentHeight = 0;
```
Comment (tomashapl) [inline src/TripSegment/index.js]: You shouldn't define initial values:

image

File: src/TripSegment/index.js [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+              </WrapperColumn>
+              <WrapperColumn>
+                <Text type="primary">{arrival}</Text>
+                <Text type="primary">{departure}</Text>
+              </WrapperColumn>
+            </StyledSegmentWrapper>
+            <WrapperCarrier>
+              <Text type="secondary" size="small">{`~${duration}`}</Text>
+              <CarrierLogo size="medium" carriers={[carrier]} />
+              <Chevrons>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: Please create separate component something like:

```
const ExpandableIcon = ({ expanded }) => (
<StyledExpandableSection>
{expanded ? <ShowLess size="small" color="secondary" /> : <ShowMore size="small" color="secondary" />}
</StyledExpandableSection>
)
```
```

File: src/TripSegment/index.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+    const { toggle, visible } = this.state;
+    const { contentHeight } = this;
+
+    return (
+      <StyledSegmentWrapper dataTest={dataTest}>
+        <Milestone onClick={this.handleToggle}>
+          <Icon type={carrier.type || CARRIER_TYPE_OPTIONS.AIRLINE} />
+          <ArrowDecor />
+        </Milestone>
+        <StyledSegment>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentContent`

File: src/TripSegment/index.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+      theme.orbit.paletteCloudNormal
+    }`};
+  transition: border-color ${({ theme }) => theme.orbit.durationFast} ease-in-out;
+`;
+
+StyledSegment.defaultProps = {
+  theme: defaultTokens,
+};
+
+const Chevrons = styled.div`
```
Comment (tomashapl) [inline src/TripSegment/index.js]: I am not sure if it looks like in design.

image

arrows are moved little bit left

File: src/TripSegment/index.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+    const { contentHeight } = this;
+
+    return (
+      <StyledSegmentWrapper dataTest={dataTest}>
+        <Milestone onClick={this.handleToggle}>
+          <Icon type={carrier.type || CARRIER_TYPE_OPTIONS.AIRLINE} />
+          <ArrowDecor />
+        </Milestone>
+        <StyledSegment>
+          <WrapperInner toggle={toggle} onClick={this.handleToggle}>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentOverviewWrapper`

File: src/TripSegment/index.js [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+            <StyledSegmentWrapper>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentOverview`

File: src/TripSegment/index.js [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+              <WrapperColumn>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentOverviewColumn`

File: src/TripSegment/index.js [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+                <Time>
+                  <Text type="attention">{departureTime}</Text>
+                </Time>
+              </WrapperColumn>
+              <WrapperColumn>
+                <Text type="primary">{arrival}</Text>
+                <Text type="primary">{departure}</Text>
+              </WrapperColumn>
+            </StyledSegmentWrapper>
+            <WrapperCarrier>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentCarrier`

File: src/TripSegment/index.js [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+              <Chevrons>
+                {toggle ? (
+                  <ShowLess size="small" color="secondary" />
+                ) : (
+                  <ShowMore size="small" color="secondary" />
+                )}
+              </Chevrons>
+            </WrapperCarrier>
+          </WrapperInner>
+          <ChildrenWrapper
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentChildrenWrapper`

File: src/TripSegment/index.js [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+arrowStyle.defaultProps = {
+  theme: defaultTokens,
+};
+
+const ArrowDecor = styled.div`
+  position: relative;
+  width: 6px;
+  height: 12px;
+  right: -1px;
+  z-index: 1;
```
Comment (tomashapl) [inline src/TripSegment/index.js]:
```
&:before, &:after {
  position: absolute;
  content: " ";
  width: 0;
  height: 0;
  border-style: solid;
  transition: border-color ${({ theme }) => theme.orbit.durationFast} ease-in-out;
}
```

File: src/TripSegment/index.js [hunk 13]
```diff
@@ -0,0 +1,329 @@
...
+    background: ${({ theme }) => theme.orbit.paletteWhite};
+    transition: color ${({ theme }) => theme.orbit.durationFast} ease-in-out;
+  }
+`;
+
+Milestone.defaultProps = {
+  theme: defaultTokens,
+};
+
+const arrowStyle = css`
```
Comment (tomashapl) [inline src/TripSegment/index.js]: Please remove this and do it in below way

File: src/TripSegment/index.js [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+      duration,
+      carrier,
+      dataTest,
+    } = this.props;
+    const { toggle, visible } = this.state;
+    const { contentHeight } = this;
+
+    return (
+      <StyledSegmentWrapper dataTest={dataTest}>
+        <Milestone onClick={this.handleToggle}>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentMilestone`

File: src/TripSegment/index.js [hunk 15]
```diff
@@ -0,0 +1,329 @@
...
+          <Icon type={carrier.type || CARRIER_TYPE_OPTIONS.AIRLINE} />
+          <ArrowDecor />
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentMilestoneArrow`

File: src/TripSegment/index.js [hunk 16]
```diff
@@ -0,0 +1,329 @@
...
+      carrier,
+      dataTest,
+    } = this.props;
+    const { toggle, visible } = this.state;
+    const { contentHeight } = this;
+
+    return (
+      <StyledSegmentWrapper dataTest={dataTest}>
+        <Milestone onClick={this.handleToggle}>
+          <Icon type={carrier.type || CARRIER_TYPE_OPTIONS.AIRLINE} />
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `MilestoneIcon`

File: src/TripSegment/index.js [hunk 17]
```diff
@@ -0,0 +1,329 @@
...
+          <ArrowDecor />
+        </Milestone>
+        <StyledSegment>
+          <WrapperInner toggle={toggle} onClick={this.handleToggle}>
+            <StyledSegmentWrapper>
+              <WrapperColumn>
+                <Time>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentOverviewTime`

File: src/TripSegment/index.js [hunk 18]
```diff
@@ -0,0 +1,329 @@
...
+          </WrapperInner>
+          <ChildrenWrapper
+            contentHeight={contentHeight}
+            toggle={toggle}
+            visible={visible}
+            innerRef={node => {
+              this.node = node;
+            }}
+          >
+            <ChildrenWrapperContent>{children}</ChildrenWrapperContent>
```
Comment (tomashapl) [inline src/TripSegment/index.js]: `StyledTripSegmentChildren`

File: src/TripSector/TripDate/index.js [hunk 19]
```diff
@@ -0,0 +1,37 @@
+// @flow
```
Comment (tomashapl) [inline src/TripSector/TripDate/index.js]: `TripDate` occurs once time in `TripSector` right ? better solution is pass `date` as prop

File: src/TripSegment/index.js (hunk 6 above)
Comment (wEprous) [reply src/TripSegment/index.js]: Yeah, I know. `16px/17px` won't work and you don't want to use decimals, so we should try to fix it afterward, not now.

File: src/TripSector/TripDate/index.js (hunk 19 above)
Comment (wEprous) [reply src/TripSector/TripDate/index.js]: `TripDate` is not used in `TripSector` at all.

**Comment by tomashapl** [comment]
@wEprous there is missing milestone line

image

**Comment by wEprous** [comment]
Fixed :)

**Comment by tomashapl** [comment]
Great job guys 👏 @mainframev @wEprous

**Pull request merged**